packages/lwc-language-server/src/javascript/__tests__/type-mapping.test.ts

```diff
@@ -0,0 +1,38 @@
+import * as path from 'path';
+import { collectBundleMetadata, BundleConfig, ScriptFile } from '@lwc/metadata';
+import { transform } from '@lwc/old-compiler';
+import { CompilerOptions as OldCompilerOptions } from '@lwc/old-compiler/dist/types/compiler/options';
+import { mapLwcMetadataToInternal } from '../type-mapping';
+import { Metadata } from '../../decorators';
+import * as fs from 'fs-extra';
+
+it('can map new metadata to old metadata', async () => {
```

Comment:
This test ensures that `mapLwcMetadataToInternal(collectBundleMetadata(filename))` produces the exact same output as `transform(...). metadata`, bug-for-bug.

As a first step, this seemed like the best way to ensure introduction of `@lwc/metadata` would not result in bugs in the language server. Over time, if the language server is refactored to leverage `@lwc/metadata`'s output more directly, this test can be relaxed or removed altogether.